util/network-devp2p/src/discovery.rs

```diff
@@ -728,7 +743,7 @@ impl<'a> Discovery<'a> {
 		trace!(target: "discovery", "Got {} Neighbours from {:?}", results_count, &from);
 		for r in rlp.at(0)?.iter() {
 			let endpoint = NodeEndpoint::from_rlp(&r)?;
```

Comment:
Won't this still fail on bootnodes?

Reply:
yes, it most likely will, but we have no way of getting the socket address here.

Reply:
@ngotchac

Reply:
I'm wondering what would happen if a node is sent here as a neighbour with a UDP port but no TCP port? It shouldn't fail, as you can still get discovery from it. If it fails, something needs to be done so it doesn't.

Reply:
My bad, I thought this would error on empty TCP port. All good then.